File: Demo1/app/src/main/java/com/example/myapplication/UpdateScreen.java
package com.example.myapplication;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.io.FileNotFoundException;
import java.io.InputStream;
import java.net.URI;

public class UpdateScreen extends AppCompatActivity implements View.OnClickListener {
    EditText name;
    ImageView image;
    Button button;
    Uri uri;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_update_screen);

        name = findViewById(R.id.editText);
        image = findViewById(R.id.imageView);
        button = findViewById(R.id.button);

        image.setOnClickListener(this);
        button.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch(v.getId()) {
            case R.id.imageView:
                Intent intent = new Intent();
                intent.setType("image/*");
                intent.setAction(Intent.ACTION_GET_CONTENT);
                startActivityForResult(Intent.createChooser(intent, "Select Picture"), 1);
                break;
            case R.id.button:
                if (name.getText().equals("")) {
                    Toast.makeText(this, "¯Please enter a name", Toast.LENGTH_SHORT).show();
                } else {
                    Intent goBack = new Intent(this, MainActivity.class);
                    goBack.putExtra("UPDATED_IMAGE", uri);
                    goBack.putExtra("UPDATED_NAME", name.getText().toString());
                    setResult(RESULT_OK, goBack);
                    finish();
                }
                break;
        }
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == 1) {
            try {
                final Uri imageUri = data.getData();
                final InputStream imageStream = getContentResolver().openInputStream(imageUri);
                final Bitmap selectedImage = BitmapFactory.decodeStream(imageStream);
                uri = imageUri;
                image.setImageBitmap(selectedImage);
            } catch (FileNotFoundException e) {
                e.printStackTrace();
                Toast.makeText(this, "¯\\_(ツ)_/¯ ", Toast.LENGTH_SHORT).show();
            }
            
        } else {
            Toast.makeText(this, "¯\\_(ツ)_/¯ ", Toast.LENGTH_SHORT).show();
        }
    }
}

File: Demo1-ViewsAndActivities-Kotlin/app/src/main/java/com/example/viewsandactivitiesdemokotlin/UpdateScreen.kt
package com.example.viewsandactivitiesdemokotlin

import android.app.Activity
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_update_screen.*
import java.io.InputStream

class UpdateScreen : AppCompatActivity(), View.OnClickListener {
    var uri : Uri? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_update_screen)

        submit.setOnClickListener(this)
        selectPic.setOnClickListener(this)
    }

    override fun onClick(v : View?) {
        // Basically works like a switch function in Java.
        when (v?.id) {
            // If the view's id is that of the submit button, start a new activity
            R.id.submit -> {
                // If the image hasn't been selected, or the name hasn't been inserted, prompt the user
                if (uri == null || editText.text.toString().isEmpty()) {
                    Toast.makeText(this, "Please select an image and enter your name", Toast.LENGTH_SHORT).show()
                } else {
                    // Create a new intent, and put in the extra data of the image Uri and the name
                    val intent: Intent = Intent(applicationContext, MainActivity::class.java)
                    intent.putExtra("NAME", editText.text.toString())
                    intent.putExtra("IMAGE", uri)
                    setResult(Activity.RESULT_OK, intent)
                    // Finish takes you back to whatever activity created this activity with
                    // the appropriate resultCode (1 in this case)
                    finish()
                }
            }
            R.id.selectPic -> {
                // Create a new content selector intent to let the user select an image
                val intent : Intent = Intent()
                intent.setType("image/*")
                intent.setAction(Intent.ACTION_GET_CONTENT)
                startActivityForResult(Intent.createChooser(intent, "Select Picture"), 1)
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        // If you successfully came back from the intent to select an image
        if (requestCode == 1) {
            // Serialize the image into an InputStream to pass to MainActivity
            val imageUri : Uri? = data?.data
            val imageStream : InputStream? = contentResolver.openInputStream(imageUri!!)
            val selectedImage : Bitmap = BitmapFactory.decodeStream(imageStream)
            selectPic.setImageBitmap(selectedImage)
            // Save the imageUri to a public variable so that we can pass it into the intent later.
            uri = imageUri
        }
    }
}
